feat: Multi Dev Portal (#7561)
Kong/docs.konghq.com: app/konnect/dev-portal/create-dev-portal.md

To create a Dev Portal, do one of the following:
    
* Create a private Dev Portal using the `/portals` endpoint:
    ```sh
    curl --request POST \
    --url https://{region}.api.konghq.com/v2/portals \
    --header 'Authorization: <personal-access-token>' \
    --header 'Content-Type: application/json' \
    --data '{
        "name": "DEV_PORTAL_NAME",
        "is_public": false,
        "rbac_enabled": true,
        "auto_approve_applications": true,
        "auto_approve_developers": true,
        "custom_domain": "api.example.com",
        "custom_client_domain": "portal.example.com",
        "default_application_auth_strategy_id": "41f3adcb-5f5b-4c97-ab08-3ac2777aa6ab"
    }'
    ```
    Be sure to replace the PAT as well as the following placeholders with your own values: 
    * `name`
    * `rbac_enabled`
    * `auto_approve_applications`
    * `auto_approve_developers`
    * `custom_domain`
    * `custom_client_domain`

>     * `default_application_auth_strategy_id`
>     Refer to the API spec for more information. 

Author: ```suggestion
    * `default_application_auth_strategy_id`
    
    Refer to the API spec for more information. 
```

---
title: Create a Dev Portal
---

> The {{site.konnect_short_name}} Dev Portal is a customizable website for developers to locate, access, and consume API services. You can create both public and private Dev Portals, depending on your use case. With a public Dev Portal, anyone with the URL can view the APIs you add to your Dev Portal. With a private Dev Portal, only those who have signed in to the Dev Portal can see the APIs you add to your Dev Portal. Additionally, when you configure private Dev Portals, you can also allow developers to register apps with your APIs. You can also publish your APIs to both a private Dev Portal as well as a public one, such as if you want to have one Dev Portal for internal developers at your company as well as allow external developers at other companies to create apps with some of your APIs.

Reviewer: ```suggestion
The {{site.konnect_short_name}} Dev Portal is a customizable website for developers to locate, access, and consume API services.

Users can create both public and private Dev Portals, depending on their use case. Public Dev Portals do not require users to login to see published APIs, and are discoverable on the internet. Private Dev Portals require users to create an account and login in order to see published APIs. Coming soon, Konnect will support Dev Portals that contain pages that are public _and_ private, allowing the public to browse the publicly available catalog, but requiring users to login to see a full catalog, and begin consuming those APIs.

When Portal admins require developers to create an account and login to use the Dev Portal, developers can "self serve" their API consumption. Developers can create applications, register them to their target APIs, and generate credentials to start consuming those APIs.
```

## Prerequisites

> Configure an authentication strategy if you plan to use one. This auth strategy will be how developers authenticate when they use your APIs. <!-- tried searching for a link to a doc for this, is it just setting up an auth plugin?-->

Reviewer: ```suggestion
Configure an [authentication strategy](https://deploy-preview-7561--kongdocs.netlify.app/konnect/dev-portal/applications/enable-app-reg/) if you plan to use one. This auth strategy will be how developers authenticate when they use your APIs. <!-- tried searching for a link to a doc for this, is it just setting up an auth plugin?-->
```

## Steps
{% navtabs %}
{% navtab Konnect UI %}
1. In {% konnect_icon dev-portal %} [**Dev Portal**](https://cloud.konghq.com/portal), click **Dev Portal** on the top right.
1. In the Create Dev Portal dialog, configure your Dev Portal settings:
    1. Enter general information for your Dev Portal, including a name and optional description.

>     1. Select if you want the Dev Portal to be private or public. Developers can only register apps to private Dev Portals.

Reviewer: ```suggestion
    1. Select if you want the Dev Portal to be private or public. Anyone can see APIs published to a public portal, only registered and logged in users can see APIs in a private portal. Note: Developers can only get credentials and start consuming APIs in a self serve way in a private Dev Portal.
```

    1. Private Dev Portal only: Configure the following settings:
        * **Auto Approve Developers**: If you enable this setting, this will allow developers to automatically be approved when they sign up for your private Dev Portal. If you don't enable this, you must configure single sign-on or manually approve developers as they register. 
        * **Auto Approve Applications**: If you enable this setting, apps that developers create with your APIs will be automatically approved. If you don't enable this, you must manually approve or deny application registrations.

>         * **Portal RBAC**: You must enable this setting if you plan to allow developers to register with your Dev Portal. Portal RBAC allows you to assign developers to teams and roles that determine if they can only view or view and consume the APIs in your Dev Portal.

Reviewer: ```suggestion
        * **Portal RBAC**: Portal RBAC allows you to assign developers to teams and roles that determine if they can only view or view and consume the APIs in your Dev Portal.
```

Reviewer: this "You must enable this setting if you plan to allow developers to register with your Dev Portal. " is not true

>         * **Default Application Auth Strategy**: Select the auth strategy you want the APIs in your Dev Portal to use if they require authentication. This auth strategy will be how developers authenticate when they use your APIs.

Reviewer: ```suggestion
        * **Default Application Auth Strategy**: Select the auth strategy you want the APIs in your Dev Portal to use if they require authentication. This auth strategy will be how developers authenticate when they register to consume your APIs. Note: Selecting a default auth strategy makes it easier for your API Product Owners to publish an API to a Dev Portal that requires authentication, as the default auth strategy will be auto selected for them. If you're not sure which auth strategy to use you can select the pre-created key-auth strategy.
```

    You should get a `201` response like the following:
    ```sh
    {
        "name": "My Dev Portal",
        "is_public": true,
        "custom_domain": "api.example.com",
        "custom_client_domain": "portal.example.com",
        "id": "4959b8cc-92fb-4197-b836-3eab20b16147",
        "created_at": "2024-06-24T16:36:01.397Z",
        "updated_at": "2024-06-24T16:36:01.397Z",
        "description": null,
        "display_name": "Developer Portal",
        "auto_approve_developers": false,
        "auto_approve_applications": false,
        "rbac_enabled": false,
        "labels": {},
        "application_count": 0,
        "developer_count": 0,
        "published_product_count": 0,
        "default_domain": "e7edf6822f8a.us.portal.konghq.com",
        "default_application_auth_strategy_id": "eae8d5ab-0378-3958-9444-ca3091541cff"
    }
    ```

{% endnavtab %}
{% endnavtabs %}

## Access the Dev Portal

You can access the Dev Portal using the Dev Portal URL. The URL is displayed when you click {% konnect_icon dev-portal %} **Dev Portal** to open the **Published API Products** overview page.
The Dev Portal URL looks like this: 
    

>     https://example.us.portal.konghq.com/

Reviewer: ```suggestion
    https://{portalId}.{region}.portal.konghq.com/
```
